rfcs/protocols/asset-exchange/corda-htlc.md

```diff
@@ -6,45 +6,49 @@
-
-#### Process
-The procedures for setting up an HTLC one one Corda network, which would need to be mirrored in the counter party network, involves the following:
-- Party A and B agree on terms of the contract (i.e. secret, time window, asset amount etc.)
-- Party A creates a transaction that takes its asset state as input and produces i) a hash lock encumbrance state and ii) the asset state with a reference to the encumbrance. Party A sends this transaction to Party B.
-- **Claiming**: Party B submits a transaction to claim the locked asset with the pre-image of the secret. If the pre-image is valid and the transaction is within the defined time window (as validated by the encumbrance contract), the output of the transaction will be a new state representing the asset with Party B as its owner that no longer has the encumbrance state attached to it.
-- **Refund**: Party A submits a transaction to claim back its locked asset. If this claim occurs after the defined window has elapsed, then a new state representing the state with Part A as the owner is output that no longer has the encumbrance state attached to it.
-
-Both claiming and refunding transactions can be performed unilaterally because during the locking phase the asset is associated with a Composite key of the two parties.
-
-![Corda HTLC](../../resources/images/cordapp-htlc.png)
-
-### Generalisability
-The hash lock encumbrance state and associated contract can be offered as reusable pieces as part of Weaver. These pieces are largely agnostic of the specifics of the application. They can further be instrumented to *emit* custom Weaver events making automation of the orchestration of cross-chain HTLCs possible.
+- Since: 21-Oct-2022
+
+## Generic Corda Network
+Below figure depicts the different components of a Corda network. Primarily it consists of the infrastructure used for the creation of the network and the distributed applications (CorDapps) deployed on the network. Any such CorDapp is a combination of contracts, flows and a client component to interact with the network.
+<img src="../../resources/images/generic-corda-network.png" width=80%>
+
+## Weaver enabled Corda Network
+Below figure depicts the different components of a Corda network with Weaver adaptations (highlighted in _green_ color). It requires changes in both infrastructure and CorDapp (contracts, flows, and client) layers. However, not all these adaptations are needed to demonstrate asset exchange protocol. For example, changes to infrastructure layer (addition of _Relays_) and changes to client layer (addition of _Identity Services_ and _Drivers_) are only needed to demonstrate _data sharing_ and _asset transfer_ protocols.
```

Comment:
"Weaver adaptations (highlighted in _green_ color). It requires changes in both infrastructure and CorDapp (contracts, flows, and client) layers." --> "Weaver-augmented capabilities, including infrastructural components like relays and drivers and CorDapps (contracts, flows, and client)."

"these adaptations" --> "of the Weaver-augmented capabilities".

Use "implement" instead of "demonstrate". The RFCs are meant to be specs, not demo guides.

Reply:
Done